kairon/shared/llm/gpt3.py

```diff
@@ -41,13 +43,30 @@ def train(self, *args, **kwargs) -> Dict:
         self.__create_collection__(self.bot + self.suffix)
         self.__create_collection__(self.bot + self.cached_resp_suffix)
         count = 0
-        contents = list(BotContent.objects(bot=self.bot))
+        search_payload = {}
```

Comment:
wrong logic. please go through this again

Reply:
also move this logic into separate smaller methods (had requested earlier too)

Reply:
is it necessory to call .__get_embedding in every if else block when it can be called only once. you just have to define the data on which you have  to create embeddings in your if else block